packages/vue/src/components/range/MultiRange.jsx

```diff
@@ -0,0 +1,252 @@
+        currentValue = [...currentValue, currentItem];
+        selectedValues = { ...selectedValues, [item]: true };
+      }
+      const performUpdate = () => {
+        this.state.currentValue = currentValue;
+        this.state.selectedValues = selectedValues;
+        this.updateQueryHandler(currentValue, props);
+        this.locked = false;
+        this.$emit('valueChange', 'divyanshu');
+      };
+
+      checkValueChange(
+        props.componentId,
+        currentValue,
+        props.beforeValueChange,
+        performUpdate
+      );
+    },
+
+    updateQueryHandler(value, props) {
+      const query = props.customQuery || MultiRange.defaultQuery;
+      this.updateQuery({
+        componentId: props.componentId,
+        query: query(value, props),
+        value,
+        label: props.filterLabel,
+        showFilter: props.showFilter,
+        URLParams: props.URLParams,
+        componentType: 'MULTIRANGE'
+      });
+    },
+
+  },
+
+  watch: {
+    react() {
+      this.setReact(this.$props);
+
+    },
+    dataField() {
+      this.updateQueryHandler(this.$data.currentValue, this.$props);
+    },
+    defaultSelected(newVal) {
+      this.selectItem(newVal);
+
+    },
+  },
+
+  created() {
+    const onQueryChange = (...args) => {
+      this.$emit('queryChange', ...args);
+    };
+
+    this.setQueryListener(this.$props.componentId, onQueryChange, null);
+  },
+
+  beforeMount() {
+    this.addComponent(this.$props.componentId);
+    this.setReact(this.$props);
+    if (this.selectedValue) {
+      this.selectItem(this.selectedValue, true);
+    } else if (this.$props.defaultSelected) {
+      this.selectItem(this.$props.defaultSelected, true);
+    }
+  },
+
+  beforeDestroy() {
+    this.removeComponent(this.$props.componentId);
+  },
+
+  render() {
+    return (
+      <Container class={this.$props.className}>
+        {this.$props.title && (
+          <Title class={getClassName(this.$props.innerClass, 'title')}>
+            {this.$props.title}
+          </Title>
+        )}
+        <UL class={getClassName(this.$props.innerClass, 'list')}>
+          {this.$props.data.map(item => {
+            const selected
+              = !!this.$data.currentValue
+              && this.$data.currentValue.label === item.label;
+            return (
+              <li key={item.label} class={`${selected ? 'active' : ''}`}>
+                <Checkbox
+                  class={getClassName(this.$props.innerClass, 'checkbox')}
+                  id={`${this.$props.componentId}-${item.label}`}
+                  name={this.$props.componentId}
+                  value={item.label}
+                  onChange={this.handleChange}
+                  type="Checkbox"
+                  checked={this.state.selectedValues[item.label]}
+                  show={this.$props.showCheckbox}
+                />
+                <label
+                  class={getClassName(this.$props.innerClass, 'label')}
```

Comment:
Is `htmlFor` missing here?

Reply:
for all vue components add till so far we have used `<label />`

Reply:
@bietkul where I have to used `htmlFor`?

Reply:
ok, it's fine if we're not using it.